cfme/tests/test_db_restore.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+
+import fauxfactory
+import pytest
+
+from cfme.configure import configuration as config
+from cfme.cloud import provider as cloud_provider
+from cfme.infrastructure.provider import wait_for_a_provider
+from cfme.infrastructure.virtual_machines import Vm
+from fixtures.pytest_store import store
+from utils.appliance import provision_appliance
+from utils.log import logger
+from utils.providers import setup_a_provider
+from utils import testgen, version
+
+pytest_generate_tests = testgen.generate(testgen.infra_providers, scope="module")
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.provider_by_type(
+        metafunc, ['virtualcenter'], choose_random=True)
+    if not idlist:
+        return
+    metafunc.parametrize(argnames, argvalues, ids=idlist, scope='module')
+
+
+@pytest.fixture(scope="module")
+def test_vm(request, provider_crud, provider_mgmt):
```

Comment:
The pytest_generate_tests also has choose_random=True,because of which the test is run on only one provider.